storage/heap/hp_record.cc

```diff
@@ -267,17 +268,19 @@ void hp_copy_record_data_to_chunkset(const HP_SHARE &info, const uchar *record,
   src_offset.
 */
 #if !defined(DBUG_OFF) && defined(EXTRA_HEAP_DEBUG)
-#define SWITCH_TO_NEXT_CHUNK_FOR_READ(share, curr_chunk, src_offset)         \
-  {                                                                          \
-    curr_chunk = *((uchar **)(curr_chunk + share->recordspace.offset_link)); \
-    src_offset = 0;                                                          \
-    dump_chunk(share, curr_chunk);                                           \
+#define SWITCH_TO_NEXT_CHUNK_FOR_READ(share, curr_chunk, src_offset) \
+  {                                                                  \
```

Comment:
Since you are touching this anyway, maybe convert to an inline function with inout reference args? Or at least do the proper do { ... } while(0) compound-statement-in-a-macro thing